convert-to-declarative-plugin/src/main/java/io/jenkins/plugins/todeclarative/converter/buildwrapper/BuildTimeoutWrapperConverter.java

```diff
@@ -19,10 +19,11 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import org.jenkinsci.plugins.variant.OptionalExtension;
 
 import static io.jenkins.plugins.todeclarative.converter.api.ModelASTUtils.buildKeyPairArg;
 
-@Extension(optional = true)
+@OptionalExtension(requirePlugins = { "build-timeout" })
```

Comment:
This is the main fix. `@Extension(optional = true)` only works if the extension fails to be loaded during extension loading, which is normally the case when the extension is extending an extension point defined in the optional dependency, but that is not the case here. I tried also adding a static initializer that used a class from the dependency, but that didn't work.